### PR TITLE
ci: upgrade actions to avoid node16 deprecation warning

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

See -

* https://github.com/actions/setup-node/issues/850
* https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/
* ... and what can be foreseen - https://github.blog/changelog/2023-05-04-github-actions-all-actions-will-run-on-node16-instead-of-node12/

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: .github/workflows/size-label.yml
Comment (tisonkun) [inline .github/workflows/size-label.yml:1]: I never found these labels help. Let's reduce such burden unless it's necessary.

**Review by sunng87 — APPROVED** [review]

File: .github/workflows/size-label.yml
Comment (waynexia) [reply .github/workflows/size-label.yml:1]: This is intended for warning xlarge PRs. But it would repeat for every single commit which is very annoying...

Comment (tisonkun) [reply .github/workflows/size-label.yml:1]: Let's handle xlarge PRs with CONTRIBUTINNG rules and trying to apply it with reviewers'/authors' consensus.

**Review by zyy17 — APPROVED** [review]
lgtm

**Pull request merged**